### ISSUE TITLE
Global contents

### ISSUE THREAD

**Issue by SpectralKH** [SpectralKH/Sylver#1]
**Link:** Global (applies to all pages)

Layout is simple and understand-able. It's white and with some lines and text sticking out, with it's own specified color.

#### Header
Same size as YouTube's, colored light-gray/white. From the right side, there's a sidebar button; it opens and closes the sidebar with a fancy animation (will make an example picture of it).
Then the logo (*/)
After that, there's a search field. Should work like any other search field; It opens a list of suggestions below, in a list, and there's a search button to the right as well.
Between the search bar and profile buttons, there's something called the "option bar".
On the right side, a “Login” button (*/login), with a “Register” button (*/register) to it's left.
After that, but with more space between the login/register buttons, an “Upload” button (*/upload).

##### Option bar
The option bar is a drop-down, containing various options for what to display on the current page, and it is only there on certain pages. It can be used to display different types of videos in a search, for example. The different pages the option bar loads will always be pre-loaded, so there's no delay if you click it (meaning they lead to bookmarks).

##### Login pop-up
The pop-up shows up right under where you click, and it's attached to the header. First off, it shows an "Email" field, and below, a "Password" field. Below the password field, there's a "Forgot password?" link (*/login/forgot). Below that, there's a "Remember me [  ]" checkbox, and a "Login" button beside.

#### Sidebar
By default being displayed on the left side. You can show the sidebar by pressing the sidebar button in the header. The sidebar is a list of different links, similar to YouTube. First, Browse (*/browse). Below it, there's "Local playlist" (*/playlist/local). It's a playlist managed freely without requiring to log in, and it's contents are stored locally. Next, we have a little spacer, followed by the title "Categories". Below it, categories will be listed.

#### Footer
The bottom includes somewhat important links and such. It starts with a logo (*/), to the left. After that, we have 1 row of links:
Customize (#Customize)
Help (*/help)
About (*/about)
Updates (*/updates)
Developers (*/developers)
Contact (*/contact)

#### "Customize" pop-up
This is basically how the pop-up will be organized. The right column will be drop-downs.
Language                            English (US)
Theme color                        Green
Sidebar position                  Left
Profile info position             Header

              Cancel           Save

Settings will be saved locally.
- Language is automatically detected to where the user's IP is.
- Theme color changes the color displayed all over the site, which is green by default.
- Changes the side the sidebar is displayed on. The sidebar button will also be moved if it's changed.
- Profile info position changes where the login/-out/register buttons are located. Can be sidebar and header.

**Comment by SpectralKH** [comment]
@forddyce Most of this can be worked on already. Also, did you notice my Skype contact request?

**Comment by forddyce** [comment]
Yes i do, please bear in mind that my time zone may be different than
yours. I live in Indonesia (GMT+8) so sometimes I might be asleep at night
when you are tying to contact me. Thanks.

**Comment by forddyce** [comment]
Just realized something, if you save something locally, then the settings will not be in multiple environment (not on mobile phone if you save from desktop for example). I suggest save the settings in database. What do you think? @SpectralKH

**Comment by SpectralKH** [comment]
@forddyce That's how it's supposed to be. Mostly, everything should be in the DB, but if you're logged out, it should be offline. Or do you think that should be hosted by us too?

**Comment by forddyce** [comment]
@SpectralKH well, if you are logged out then the data will not be used and back to the default template. I thought you wanted to save settings on user's computer locally like using a cookie.

**Comment by SpectralKH** [comment]
@forddyce I do want to save data on a user's computer, but only/mostly if you are not logged in. There is a "local playlist", which should only be saved on the user's computer. Some specific features are for only 1 computer, and I think that should be saved offline. Then you could also use incognito mode to get back to the default settings. What do you think about that?

**Comment by forddyce** [comment]
@SpectralKH I'm going to need a lot of detail on what data will be saved locally on user's computer. 

For example, you said "local playlist". Technically a web browser cannot access directly to folders and files on user's computer, so a user must somehow upload it somewhere outside (youtube and soundcloud have their own domain just for storing these assets) for the web to be able to access the local playlist.

**Comment by SpectralKH** [comment]
@forddyce Ok, I will do that. I'm not 100% sure how cookies work, but assuming they can store basic information; The local playlist would need to have video IDs listed, and in what position they would be. The playlist would only be accessible from that specific computer, then. That should be possible through cookies?

**Comment by SpectralKH** [comment]
@forddyce @bluckjigar @iyass09 I would like all of you to commit often, and work together on this one. For example, Forddyce can do the header, Bluckjigar the sidebar and Isyass the footer. I want to see how this works firstly, and for every time you work on this remember to submit it here.

**Comment by forddyce** [comment]
is it going to be pure html first?

**Comment by SpectralKH** [comment]
PHP should work fine.
